lib/scholar/linear/bayesian_ridge_regression.ex

```diff
@@ -0,0 +1,320 @@
+defmodule Scholar.Linear.BayesianRidgeRegression do
+  require Nx
+  import Nx.Defn
+  import Scholar.Shared
+
+  @derive {Nx.Container,
+           containers: [:coefficients, :intercept, :alpha, :lambda, :sigma, :rmse, :iterations, :scores]}
+  defstruct [:coefficients, :intercept, :alpha, :lambda, :sigma, :rmse, :iterations, :scores]
+
+  opts = [
+    iterations: [
+      type: :pos_integer,
+      default: 300,
+      doc: """
+      Maximum number of iterations before stopping the fitting algorithm.
+      The number of iterations may be lower is parameters converge.
+      """
+    ],
+    sample_weights: [
+      type:
+        {:or,
```

Comment:
I would allow `sample_weights` to be only `type: {:custom, Scholar.Options, :weights, []}`.